libnmstate/nm/applier.py

```diff
@@ -381,14 +381,19 @@ def _build_connection_profile(iface_desired_state, base_con_profile=None):
     if bond_opts:
         settings.append(bond.create_setting(bond_opts))
     elif iface_type == bridge.BRIDGE_TYPE:
-        bridge_options = iface_desired_state.get(bridge.BRIDGE_TYPE, {}).get(
-            LB.OPTIONS_SUBTREE
+        bridge_state = iface_desired_state.get(bridge.BRIDGE_TYPE, {})
+        bridge_options = bridge_state.get(LB.OPTIONS_SUBTREE)
+        linux_bridge_setting = bridge.create_setting(
+            bridge_options,
+            base_profile,
+            enable_vlan_filtering=any(
```

Comment:
These code should move to `bridge.create_setting()` where it could determine whether to enable VLAN filtering based on `bridge_options`. The `nm/applier` should not take care such details.

Reply:
I don't think it's possible to decide if vlan filtering should be enabled or not based on the bridge options: it's information that is defined on the bridge ports. How do you suggest we do that ? 

As I see it, I'll have to update the function's signature. Thus, might as well be explicit.

Reply:
The point is that the `applier` module does not know about such details.
It only knows who to call with what info... the code structure asks to push the details down the stack as much as possible, leaving the upper layers cleaner and generic.

I am also unsure if this check is even correct. Bridge options may be changed without specifying any vlans, but that is not to say that there are no current such ports defined. (unless you handle this as part of the "merge" step).

Reply:
I possibly am not seeing the forest because of all the trees, but the criteria to enable `vlan_filtering` on the bridge should be if there's any port that has a vlan configuration. Do we agree on that ?

What if my port_list features elements both from the desired state **and** current state ? 
I think that would allow bridge options update (without port vlan config changes) to preserve the old state .

Reply:
Decided to grab the current vlan_filtering value from the bridge_setting object.

This would preserve the value on a bridge option update.

Reply:
In my opinion this should be done inside `create_setting()`.

Reply:
1. As mentioned before, the logic needs to be passed down the stack to the nm.bridge module.
2. The only place that seems to cover all cases is to detect if vlan-filtering is to be enabled or not at the metadata stage, where you have the desired state, the current state and the ability to place metadata on the state that mentions this flag (if vlans exist on the bridge or not).
The other alternative is to query the bridge configuration in nm.bridge*, which may also work. Pick your favorite and lets see the result.

Reply:
Implemented it in the metadata section.